### PR TITLE
testimonial: added carousel on landing page

### PR DESCRIPTION
**Pull request by thelostone-mc**
**Changes:** 
- created a fadein-fadeout carousel for testimonial on the landing page
- populated with dummy content
- responsive design
- used font-awesome icons and shapes instead of the images

Refs: https://github.com/gitcoinco/web/issues/63

@tra38 I rewrote the code base using yours as a starting point. Any thoughts on this approach ? 
@owocki Also , the design laid out by @taurean was pretty sweet , but I decided to use font-awesome icons and basic shapes to recreate the design.

Could either of you spare a few min and make sure I've gotten everything right ?

Reason : 
- images are heavier 
- resizing divs and svg icons is a lot easier that images and they don't loose clarity

**files added:** 
- `carousel.css`
-  `carousel.js`
-  `testimonials.js`

**files changed:**
- `index.html`
- `footer_scripts.html`
- `head.html`

### CODE REVIEW

**Comment by owocki** [comment]
https://github.com/gitcoinco/web/pull/69
https://github.com/gitcoinco/web/pull/80
https://github.com/gitcoinco/web/pull/79

@tra38 @ethikz @thelostone-mc -- im thrilled that everyones excited about this and that we've got working testimonial modules, but i wish we had all coordinated better.  it makes me sad we can't use all three 3 PRs.    but i pulled down all 3 codebases and all 3 work, so that's pretty cool!

ill own not having a better claim tool.  that's on gitcoin.

so... where do we go from here?  whats best for the repo?

**Comment by taurean** [comment]
looks like #80 is the closest to the mockup at the moment. the prev/next controls still need to be centered properly within the circle. The icons surrounded the user-photo is lacking that layered effect with the illustrative elements but that might be because that place-holder image is transparent? In either case, there should probably be a background color set to the image to mitigate that issue. The cited name should also be in caps. Thanks everyone!

**Comment by tra38** [comment]
>so... where do we go from here? whats best for the repo?

Since open source is built on collaboration, it would seem the best way of handling this scenario is to merge all the best ideas from the different PRs into a single PR to be merged into master. All people involved will share in the bounty, based on how much contribution each person did to the overall task.

I'm perfectly fine with @thelostone-mc rewriting my codebase. There's some features in the old codebase that I want to port over to the new one (such as allowing programmers to change slide timings), mostly by adding commits to this PR. Would that be okay with you, @thelostone-mc?

**Comment by thelostone-mc** [comment]
@taurean 
- Ah I didn't notice the mis-alignment on my machine. That looks ugly, will fix that 
- Yeah the layered effect isn't visible cause the `avatar` has a transparent background.I tested it out with a photo with a solid background, no issues there. We could set an overlay color, but I'm not sure if css has the ability to detect if the background is transparent. Any thoughts on this @owocki , @tra38  ?
- will change the cited name to be in caps 

@tra38 By all means -> port it over via a commit :D  (For now, one has to edit this [line](https://github.com/gitcoinco/web/pull/80/files#diff-cdd7ab1176407284461646510fcf0c55R5) here to change the timings )

**Comment by taurean** [comment]
awesome @thelostone-mc. re: background image, I don't think you need to detect transparency first. setting a background color of #FAFAFA or something to `img` or the parent containing img and making sure it doesn't peak out underneath opaque images should work just fine. I can whip a codepen example up if thats helpful to illustrate what I mean.

**Comment by thelostone-mc** [comment]
@taurean 
1) Setting the background to #FAFAFA 
screen shot 2017-12-05 at 1 39 17 pm

2) Setting the background to #FFF looks neater but then the layering effect gets compromised to an extent  (Seems like the best option )
screen shot 2017-12-05 at 1 40 59 pm

3) Setting the background of the parent to #FAFAFA looks perfect but it messes up when you have a photo with a non-transparent bg. (Ignore the positions of the circles for now )

**Good:**
screen shot 2017-12-05 at 1 53 50 pm

**Bad:**
screen shot 2017-12-05 at 1 54 20 pm

PS: Is the font-size for the cited name alright or does it need to be increased ?

**Comment by owocki** [comment]
thanks for working together everyone.. im thrilled that you're working together in the spirit of open source :)

let me figure out what to do about the bounty payout here while you all iron out the PR.

**Comment by owocki** [comment]
@ethikz what do you think of this approach?

**Comment by tra38** [comment]
@thelostone-mc, here's [my commit](https://github.com/tra38/web/commit/18eeabb2ed42c5ecf6faaaa83cbf499cefeca914) with my suggested changes to your code. It's in a separate branch since I don't know how to push code onto your branch. Let me know how you like it.

**Comment by taurean** [comment]
@thelostone-mc here is an [example codepen](https://codepen.io/taurean/pen/pdBGZq) of what I'm talking about, there shouldn't be any additional padding causing the issue in the screenshot I don't think. re: font-size I can't inspect that image to check but the computed size that's rendered for the block quote is 20px and the cited name is 16px. Both the block quote and cited name should be a medium weight so probably like `font-weight: 600` if I had to guess?

**Comment by taurean** [comment]
@owocki do you happen to know if the testimonial images will be photos or illustrations? Maybe we're thinking too much about setting fallbacks if we know the exact images we're building for.

**Comment by thelostone-mc** [comment]
@tra38 Gimme a day! Let me fix the css up a bit and squash the commit, then just send out a PR to my branch , i'll merge it and it would be reflected here ( sound good ? )

@taurean so basically the first photo I'd posted ^_^ will do

**Comment by owocki** [comment]
i'm still gathering the images.  do you think gitcoin style illustrations (like my avatar) would look good?  if so, we could probably get them hooked up

**Comment by thelostone-mc** [comment]
@tra38 I''ve made the changes (feel free to double check ^_^ )
Could you send out the PR ? I'll merge it and we'll get this PR set in motion :D

**Comment by tra38** [comment]
@thelostone-mc: PR sent! 👍

**Comment by thelostone-mc** [comment]
@owocki I guess we are done from our end ^_^

**Comment by owocki** [comment]
sawwwweetttt.. rdy for me to test then? `thelostone-mc:testimonial` is the branch i want to test?

**Comment by thelostone-mc** [comment]
@owocki That's the one \m/

**Comment by owocki** [comment]
pulled it down to QA.

1. looks very nice.  good work yall!
2. can the cursor turn to a opinter when you hover over the left/right arrows?
3. responsive works very well on chrome and on safari!

woo hoo!

File: app/assets/v2/css/carousel.css [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+
+.carousel .avatar .closed {
+  position: relative;
+  top: -38px;
+  left: 98px;
+  font-size: 8px;
+}
+
+.carousel .avatar .circle-small {
+	position: relative;
```
Comment (owocki) [inline app/assets/v2/css/carousel.css]: inconsistent tab / spacing on this file.  can we clean that up real quick?

File: app/retail/views.py [hunk 2]
```diff
@@ -26,7 +26,14 @@
 
 
 def index(request):
+    slides = [
```
Comment (owocki) [inline app/retail/views.py]: i love that this is abstracted out of the template :)

File: app/retail/views.py [hunk 3]
```diff
@@ -26,7 +26,14 @@
...
+        ("Kevin Owocki","/static/v2/images/avatar.png", "This could be the next freelance boom."),
+        ("Aditya Anand","/static/v2/images/avatar.png", "I feel it is so awesome to have the opportunity through Gitcoin to do what I love and get paid for it, and to have reasonable freedom about the way I work, that it already seems too good to be true."),
+        ("Kevin Owocki","/static/v2/images/avatar.png", "Staying lean, building the community, iterating... The way to go vs many projects that are just looking for the money grab immediately.")
+    ]
     context = {
+        'slides': slides,
+        'slideDuration': 6000,
```
Comment (owocki) [inline app/retail/views.py]: # ms

**Comment by owocki** [comment]
just left a few comments.  then i think this is rdy to merge!

**Comment by taurean** [comment]
chiming in really quick, looks like the controls for navigating to other testimonials are divs, ideally these should be buttons for accessibility reasons.

File: app/assets/v2/css/carousel.css (hunk 1 above)
Comment (thelostone-mc) [reply app/assets/v2/css/carousel.css]: Haha getting that done !!  Thanks for pointing that out ^_^

File: app/retail/views.py (hunk 2 above)
Comment (thelostone-mc) [reply app/retail/views.py]: credits go to @tra38  :D

File: app/retail/views.py (hunk 3 above)
Comment (thelostone-mc) [reply app/retail/views.py]: Yup! we could rename it to `slideDurationInMs` or leave a comment -> take your pick

**Comment by thelostone-mc** [comment]
@owocki 
- re-indented the css file
- not sure if i should rename the timer variable to `slideDurationInMs`

> can the cursor turn to a opinter when you hover over the left/right arrows?

Yup!
>responsive works very well on chrome and on safari!

and firefox \m/

@taurean I meant to do that. Slipped my mind!  Fixed ^_^

**Comment by ethikz** [comment]
@owocki Sorry I didn't get any notification 😞 

What approach are you referring to?

File: app/retail/views.py (hunk 3 above)
Comment (owocki) [reply app/retail/views.py]: up to you

**Comment by owocki** [comment]
> What approach are you referring to?
@ethikz 
the one on this PR.  there was a mix up and multiple people workedo n this. see => https://github.com/gitcoinco/web/pull/80#issuecomment-349183346

**Comment by thelostone-mc** [comment]
@owocki changes done ^_^

**Comment by ethikz** [comment]
@owocki Oh...yea I saw.  Whichever one is better I'd use.  I threw it together quickly so maybe @thelostone-mc was able to provide a better solution

**Comment by owocki** [comment]
i have a few testimonials sourced and i am just circling back with everyone to make sure they're double extra secret sure they're okay with being quoted publicly.  i hope to have that content added to the PR by EOW

**Comment by owocki** [comment]
ok ive got the testimonials sourced.  testing for merge now

File: app/assets/v2/css/carousel.css (hunk 1 above)
Comment (owocki) [reply app/assets/v2/css/carousel.css]: still looks inconsistent to me

**Comment by owocki** [comment]
https://gitcoin.co <= hows it look 

heres the commit for adding the real testimonials 729710f3595e73c51735908d1fd58dd159705616

**Comment by owocki** [comment]
thanks yall :)

**Comment by ethikz** [comment]
@owocki Looks good!  Minor issue in the nav, arrows aren't exactly centered but other than that works and looks great

**Pull request merged**